web/src/topic_link_util.ts

```diff
@@ -37,13 +37,25 @@ export function html_escape_markdown_syntax_characters(text: string): string {
     return text.replaceAll(invalid_stream_topic_regex, escape_invalid_stream_topic_characters);
 }
 
-export function get_fallback_markdown_link(stream_name: string, topic_name?: string): string {
+export function get_fallback_markdown_link(
+    stream_name: string,
+    topic_name?: string,
+    message_id?: string,
+): string {
     const stream = stream_data.get_sub(stream_name);
     const stream_id = stream?.stream_id;
     assert(stream_id !== undefined);
     const escape = html_escape_markdown_syntax_characters;
     if (topic_name !== undefined) {
-        return `[#${escape(stream_name)}>${escape(topic_name)}](${internal_url.by_stream_topic_url(stream_id, topic_name, () => stream_name)})`;
+        const stream_topic_url = internal_url.by_stream_topic_url(
+            stream_id,
+            topic_name,
+            () => stream_name,
+        );
+        if (message_id !== undefined) {
+            return `[#${escape(stream_name)}>${escape(topic_name)}@💬](${stream_topic_url}/near/${message_id})`;
+        }
+        return `[#${escape(stream_name)}>${escape(topic_name)}](${stream_topic_url})`;
```

Comment:
We want spaces around the `>`, just like with the existing stream/topic link syntax.